lib/media.js

```diff
@@ -160,91 +159,86 @@ function injectTags(fileArray, site, tag) {
 
 /**
  * Gets a list of all css and js needed for the components listed.
- *
- * NOTE: the getStyles and getScripts are memoized using all arguments
- *
- * @param {Array} componentList
- * @param {string} slug
+ * @param {Object} state
  * @returns {{styles: Array, scripts: Array}}
  */
-function getMediaMap({ _components:componentList, locals }) {
+function getMediaMap(state) {
   return {
-    styles: flatMap(componentList, (component) => module.exports.getStyles(component, locals.site)),
-    scripts: flatMap(componentList, (component) => module.exports.getScripts(component, locals.site))
+    styles: module.exports.getStyleFiles(state),
+    scripts: module.exports.getScriptFiles(state)
   };
 }
 
 /**
- * Get the proper scripts for the component
+ * Get the proper styles for the component. If a styleguide is configured, use
+ * that css. Defaults to the site slug
  *
- * @param {string} name
- * @param {object} site
+ * @param {Object} state
  * @returns {array}
  */
-function getScripts(name, site) {
-  const slug = site.slug,
+function getStyleFiles(state) {
+  const { site } = state.locals,
     assetDir = site && site.assetDir || MEDIA_DIRECTORY,
     assetPath = site && site.assetPath || '',
-    assetHost = site && site.assetHost || '';
-
-  return _.map(_.filter([
-    path.join(assetDir, 'js', `${name}.js`),
-    path.join(assetDir, 'js', `${name}.${slug}.js`),
-    path.join(assetDir, 'js', `${name}.client.js`)
-  ], files.fileExists), pathJoin(assetHost, assetPath, assetDir));
-}
+    assetHost = site && site.assetHost || '',
+    siteStyleguide = site && site.styleguide;
+  let cssFilePaths;
+
+  // legacy behavior: load <component>.css and <component>.<site>.css
+  if (!site.styleguide) {
+    const siteSlug = site && site.slug;
+
+    cssFilePaths = state._components
+      .reduce((acc, curr) => {
```

Comment:
what is the var `acc` referring to?

Reply:
Pretty sure it's the local var for the accumulation of the reduction, could probably change this to `pathsToRetrieve` or something if that's more clear